src/Collection.php

```diff
@@ -5,23 +5,44 @@
 use Aeviiq\Collection\Exception\InvalidArgumentException;
 use Aeviiq\Collection\Exception\LogicException;
 
+/**
+ * @psalm-template TKey as array-key
+ * @psalm-template TValue
+ * @phpstan-template TKey
+ * @phpstan-template TValue
+ *
+ * @psalm-implements CollectionInterface<TKey, TValue>
+ * @phpstan-implements CollectionInterface<TKey, TValue>
+ */
 class Collection implements CollectionInterface
 {
     /**
+     * @psalm-var array<TKey, TValue>
+     * @phpstan-var array<TKey, TValue>
+     *
      * @var array<string|int, mixed>
      */
     private $elements;
 
     /**
+     * @psalm-var class-string<\ArrayAccess>|string
+     * @phpstan-var class-string<\ArrayAccess>|string
+     *
      * @var string
      */
     private $iteratorClass;
 
     /**
+     * @psalm-param array<TKey, TValue> $elements
+     * @phpstan-param array<TKey, TValue> $elements
+     *
+     * @psalm-param class-string<\ArrayAccess>|string $iteratorClass
+     * @phpstan-param class-string<\ArrayAccess>|string $iteratorClass
+     *
      * @param array<string|int, mixed> $elements
-     * @param string  $iteratorClass
+     * @param string $iteratorClass
      */
-    public function __construct(array $elements = [], string $iteratorClass = \ArrayIterator::class)
+    final public function __construct(array $elements = [], string $iteratorClass = \ArrayIterator::class)
```

Comment:
Made this final cause the following phpstan message:

```
  331    Unsafe usage of new static().                                                                                                                                                    
         💡 Consider making the class or the constructor final.   
```

Alternatively we can change new static() to new self() on line 367 and remove the unit tests that test this behaviour